internal/server/middleware/grpc/middleware.go

```diff
@@ -245,16 +245,10 @@ func CacheUnaryInterceptor(cache cache.Cacher, logger *zap.Logger) grpc.UnarySer
 }
 
 // AuditUnaryInterceptor sends audit logs to configured sinks upon successful RPC requests for auditable events.
-func AuditUnaryInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
+func AuditUnaryInterceptor(logger *zap.Logger, authGetter interface {
+	GetAuthenticationByID(ctx context.Context, id string) (*authrpc.Authentication, error)
```

Comment:
small nit: could we make this an unexported type right above this instead of inline? I find the inline version hard to read

```go
type authGetter interface {
  GetAuthenticationByID(ctx context.Context, id string) (*authrpc.Authentication, error)
}
```

```go
func AuditUnaryInterceptor(logger *zap.Logger, authGetter authGetter)
```

Reply:
I would go one more and make it exported. Because it is on an exported function.